### PR TITLE
feat(acp): per-turn usage reporting and live context window updates

### LINKED ISSUE #1038: Usage Reporting in Kimchi ACP (kimchi-native-chat #03)
## What to build

Kimchi's ACP server populates `PromptResponse.usage` with per-turn token totals (v1 only) and emits `UsageUpdate` sessionUpdates with context window usage during active turns (stable v1→v2). AO's chat UI shows token accounting and context window indicators for Kimchi sessions.

## Per-turn totals (`PromptResponse.usage`)

Pi-mono chains multiple `agent.prompt`/`agent.continue` calls per turn — each produces an `AssistantMessage` with its own `usage` object. Accumulate usage during the turn. In `finalizeTurn()`, emit the sum as `PromptResponse.usage`:

| ACP v1 field | Source (pi-ai `Usage`) | Required? |
|---|---|---|
| `inputTokens` | sum of `input` | required |
| `outputTokens` | sum of `output` | required |
| `cachedReadTokens` | sum of `cacheRead` | optional |
| `cachedWriteTokens` | sum of `cacheWrite` | optional |
| `thoughtTokens` | sum of `reasoning` (when provider reports it) | optional |

When no usage data was collected (e.g., cancelled before first message), `usage` is omitted.

## Live context window (`UsageUpdate` sessionUpdate)

Emit a `usage_update` sessionUpdate on turn events that carry usage, using `session.getContextUsage()` which returns `ContextUsage | undefined`. Map to ACP `UsageUpdate` as `{ size: contextWindow, used: tokens }`. **Skip emission** when `getContextUsage()` returns `undefined` or `tokens` is `null`. Do not emit on `message_update` (streaming deltas).

## Acceptance criteria

- [ ] The ACP server accumulates pi-ai `Usage` across the turn chain (input, output, cacheRead, cacheWrite, reasoning) in the `TurnContext`.
- [ ] `finalizeTurn()` returns `PromptResponse` with `usage` set to the summed totals. When no usage data was collected, `usage` is omitted.
- [ ] The ACP server emits a `usage_update` sessionUpdate on turn events, mapped from `session.getContextUsage()`.
- [ ] No `usage_update` is emitted when `getContextUsage()` returns `undefined` or `tokens` is `null`.
- [ ] No `usage_update` is emitted during streaming (`message_update` events).
- [ ] Unit tests verify: scripted events with `usage` fields produce `usage_update` sessionUpdates with correct `size`/`used` values; `getContextUsage()` returning `undefined` or `tokens: null` produces no `usage_update`.

### PR DESCRIPTION
**Pull request by cast-vytautas**
## Linked issue

Closes #1038

## What does this PR do?

Implements ticket #03 — Usage Reporting in Kimchi ACP. The ACP server accumulates pi-ai `Usage` across the turn chain and emits per-turn totals in `PromptResponse.usage` (v1). Emits `usage_update` sessionUpdates mapped from `session.getContextUsage()` on `message_end` events. Skips emission when context usage is unavailable or during streaming. Hardened against provider edge cases (NaN, undefined tokens, missing fields).

## Checklist

- [x] I have read [CONTRIBUTING.md](../CONTRIBUTING.md) and agree to the CLA
- [x] This PR links to an open issue above
- [x] Tests pass locally (`pnpm run test`)
- [x] Lint passes (`pnpm run check`)
- [x] Documentation updated if behavior changed

### CODE REVIEW

File: src/modes/acp/server.ts [hunk 1]
```diff
@@ -951,6 +1005,26 @@ export class KimchiAcpAgent implements Agent {
...
+	 * Emits a usage_update sessionUpdate mapped from
+	 * `session.getContextUsage()` → ACP UsageUpdate { size, used }. Skipped
+	 * when the session reports no estimate (undefined) or an unknown token
+	 * count (tokens: null, e.g. right after compaction) — ACP requires both
+	 * fields, so there is no null-safe emission.
+	 */
+	private sendUsageUpdate(sessionId: string, entry: SessionRecord): void {
+		const ctx = entry.session.getContextUsage()
+		if (!ctx || ctx.tokens === null) return
+		this.send({
```
[the remark this answers is not in the document]
Comment (cast-vytautas) [reply src/modes/acp/server.ts]: Addressed in commit 3f10ffcb: `sendUsageUpdate` now wraps `getContextUsage()` in try/catch — a throw is logged to stderr and returns early, so it can no longer abort turn processing on the synchronous event-emitter path.

File: src/modes/acp/server.ts [hunk 2]
```diff
@@ -951,6 +1005,26 @@ export class KimchiAcpAgent implements Agent {
...
+			sessionId,
```
[the remark this answers is not in the document]
Comment (cast-vytautas) [reply src/modes/acp/server.ts]: Addressed in commit 3f10ffcb: the guard now skips `null`, `undefined`, AND `NaN` — `if (!ctx || typeof ctx.tokens !== "number" || !Number.isFinite(ctx.tokens)) return`. Stricter than `== null` since ACP requires a real number for `used`.

File: src/modes/acp/server.ts [hunk 3]
```diff
@@ -666,6 +693,33 @@ export class KimchiAcpAgent implements Agent {
...
+			case "message_end": {
+				if (!turn) return
+				const msg = event.message
+				if (msg.role !== "assistant") return
+				const usage = msg.usage
+				if (usage) {
+					turn.usage.input += usage.input
+					turn.usage.output += usage.output
+					turn.usage.cacheRead += usage.cacheRead
+					turn.usage.cacheWrite += usage.cacheWrite
```
[the remark this answers is not in the document]
Comment (cast-vytautas) [reply src/modes/acp/server.ts]: Addressed in commit 3f10ffcb: all five usage fields accumulate with `|| 0` (undefined/NaN defaults to 0), so one malformed message cannot poison the turn totals; `reasoning` is gated on `typeof === "number" && Number.isFinite`.

**Review by Ivaka — APPROVED** [review]

**Review by vmizg — APPROVED** [review]
LGTM, just need conflict resolution
